core/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala

```diff
@@ -216,6 +216,19 @@ class SQLPlanParserSuite extends BaseTestSuite {
     }
   }
 
+  test("test subexecutionId mapping to rootExecutionId") {
+    val eventlog = ToolTestUtils.getTestResourcePath("" +
+        "spark-events-qualification/db_subExecution_id.zstd")
+    val app = createAppFromEventlog(eventlog)
+    // In Spark 3.4.0+ and later, all the sub-executions will be grouped if they are part of the
+    // the same root execution.
+    if (ToolUtils.isSpark340OrLater()) {
+      assert(app.sqlIdToInfo.values.exists(_.rootExecutionID.isDefined))
+    } else {
+      assert(app.sqlIdToInfo.values.forall(_.rootExecutionID.isEmpty))
+    }
+  }
+
```

Comment:
The test should be executed conditionally for the correct target versions. we have some unit tests that executes conditionally according to the spark version used in the unit-tests.

I do not find the test is doing anything valuable.
It is not validating the values are correct and it does not validate the SQL durations is not duplicate.

Reply:
Updated the test. Now it checks for subExecutionId's for a given rootId and also verifies that we are not double counting the durations.